src-self-hosted/translate_c.zig

```diff
@@ -2735,6 +2735,9 @@ fn transCPtrCast(
     if (ZigClangType_isVoidType(qualTypeCanon(child_type))) {
         // void has 1-byte alignment, so @alignCast is not needed
         try ptrcast_node.params.push(expr);
+    } else if (typeIsOpaque(rp.c, qualTypeCanon(child_type), loc)) {
+        // For opaque types a ptrCast is enough
```

Comment:
Should/could we assume max C alignment? (whatever alignment malloc has? or whatever `std::max_align_t` is on the target?)

Reply:
We play it safe here as the alignment for `@opaqueType` is 1, every other cast has to be vetted by the user as `alignCast` is involved.